lib/rules/no-side-effects-in-computed-properties.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * @fileoverview Don't introduce side effects in computed properties
+ * @author Michał Sajnóg
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function parseMemberExpression (node) {
+  const members = []
+  let memberExpression
+
+  if (node.type === 'MemberExpression') {
+    memberExpression = node
+
+    while (memberExpression.type === 'MemberExpression') {
+      if (memberExpression.property.type === 'Identifier') {
+        members.push(memberExpression.property.name)
+      }
+      memberExpression = memberExpression.object
+    }
+
+    if (memberExpression.type === 'ThisExpression') {
+      members.push('this')
+    } else if (memberExpression.type === 'Identifier') {
+      members.push(memberExpression.name)
+    }
+  }
+
+  return members.reverse()
+}
+
+function getComputedProperties (componentProperties) {
+  const computedPropertiesNode = componentProperties
+    .filter(p =>
+      p.key.type === 'Identifier' &&
+      p.key.name === 'computed' &&
+      p.value.type === 'ObjectExpression'
+    )[0]
+
+  if (!computedPropertiesNode) { return [] }
+
+  const computedProperties = computedPropertiesNode.value.properties
+
+  return computedProperties
+    .map(cp => {
+      const key = cp.key.name
```

Comment:
@michalsnik `cp.key` can be undefined if you are using [`experimentalObjectRestSpread`](https://babeljs.io/docs/plugins/transform-object-rest-spread/)
Solution: `if cp.type === 'Property'`

```
TypeError: Cannot read property 'name' of undefined
```

example:
```js
export default {
  computed: {
    ...mapGetters({})
  }
}
```